### PR TITLE
Add ability to initialize a replica using pglogical logical replication

### PR DESCRIPTION
**Pull request by joshraker**
Only sets up replication for the `public` schema on the `db` Database using the `aptible` user's credentials. The supported use case is Database upgrades and replication should probably be torn down once the upgrade is complete to avoid issues with logical replication in the future. Users that want to use logical replication for other reasons should configure it themselves.

- [x] Can `pglogical` be added to the default `shared_preload_libraries`? Yes
- [x] Do we want to make the default `wal_level` `logical`? Physical replication will still work. Yes

### CODE REVIEW

File: templates/etc/postgresql/PG_VERSION/main/postgresql.conf.template [hunk 1]
```diff
@@ -35,7 +35,7 @@ shared_buffers = 128MB
 # REPLICATION
 #------------------------------------------------------------------------------
 
-wal_level = hot_standby
+wal_level = logical
```
Comment (joshraker) [inline templates/etc/postgresql/PG_VERSION/main/postgresql.conf.template:38]: Logical decoding was introduced in 9.4 so this is causing 9.3 to break.

**Comment by joshraker** [comment]
I was banging my head against the wall trying to figure out why the 9.4-contrib and 9.5-contrib `config.mk` files weren't in the diff. It's because they're symbolic links to the `config.mk` files in 9.4 and 9.5 respectively. I updated the other contrib images to match but I think we should consider just duplicating the file to avoid confusion in the future.

File: bin/run-database.sh [hunk 2]
```diff
@@ -144,6 +157,52 @@ elif [[ "$1" == "--initialize-from" ]]; then
...
+  # Nodes can have multiple connection interfaces
+  psql "$2" --command "SELECT pglogical.create_node(node_name := 'aptible_publisher', dsn := '${PUBLISHER_DSN}')" \
+    || { echo "Error: Failed to create publisher node. Is there already a pglogical node on the ${database} database?" && exit 1; }
+  gosu postgres psql --dbname "${DB}" --command "SELECT pglogical.create_node(node_name := 'aptible_subscriber', dsn := '${SUBSCRIBER_DSN}')"
+
+  psql "$2" --command "SELECT pglogical.create_replication_set(set_name := '${REPLICATION_SET_NAME}')"
+  psql "$2" --command "SELECT pglogical.replication_set_add_all_tables('${REPLICATION_SET_NAME}', ARRAY['public'])"
+  psql "$2" --command "SELECT pglogical.replication_set_add_all_sequences('${REPLICATION_SET_NAME}', ARRAY['public'])"
+  gosu postgres psql --dbname "${DB}" --command "SELECT pglogical.create_subscription(subscription_name := 'aptible_subscription', provider_dsn := '${PUBLISHER_DSN}', replication_sets := ARRAY['${REPLICATION_SET_NAME}'])"
+  gosu postgres /etc/init.d/postgresql stop
```
Comment (UserNotFound) [inline bin/run-database.sh:204]: We need to `SELECT pglogical.wait_for_subscription_sync_complete('aptible_subscription');` before we stop postgres here.

File: bin/run-database.sh [hunk 3]
```diff
@@ -144,6 +157,52 @@ elif [[ "$1" == "--initialize-from" ]]; then
     echo "${TRIGGER}" >> "${DATA_DIRECTORY}/recovery.conf"
   fi
 
+elif [[ "$1" == "--initialize-from-logical" ]]; then
+  [ -z "$2" ] && echo "docker run -it aptible/postgresql --initialize-from-logical postgresql://..." && exit 1
```
[the remark this answers is not in the document]
Comment (joshraker) [reply bin/run-database.sh:161]: Done

File: bin/run-database.sh (hunk 2 above)
Comment (joshraker) [reply bin/run-database.sh:207]: Done

**Pull request merged**